### ISSUE TITLE
feat(@chill-viking/layout): use template ref to inject sections

### ISSUE THREAD

**Issue by peterjokumsen** [chill-viking/ng-libs#16]
# Purpose

Add functionality for consumer applications to specify templates for sections inside of the layout without needing to add styles for their placement.

Should use directives as the placeholder of templates, to enable type safety for data and easier to pull out of content in the base layout component.

Look and feel elements to follow later, for now focusing on placement and forcing responsive layout.

## Primary sections to add

- [x] [Header](#header-section) - #21 and #33
- [x] [Footer](#footer-section) - #35 and #40
- [ ] [Body](#body-section) - #36 (marked as wontfix, as no real need for a specific template at this time)

### Header section

Header should be displayed at the top of the layout. With the following specifications:

- Expect a title to be passed in either as an observable or a string
- Should render in the top of the page spanning the full width
- Be a `div` with a class of `cv-layout-header`
- When a header template is specified in content, the template should be rendered inside of `.cv-layout-header`
- When no header template is specified and `title` input defined in layout, render an `h1` element inside of `.cv-layout-header`
- When no header template is specified and `title` input is not defined in layout, should not render `.cv-layout-header`

### Footer section

Footer should be displayed at the bottom of the layout. With the following specifications:

- No data expected at this time
- Should render in the bottom of the page spanning the full width
- Be a `div` with a class of `cv-layout-footer`
- When a footer template is specified in content, the template should be rendered inside of `.cv-layout-footer`
- When no footer template is specified, should not render `.cv-layout-footer`

### Body section

Body should be displayed between [header](#header-section) and [footer](#footer-section) respectively. With the following specifications:

- No data expected at this time
- Should render between header and footer without margins, spanning the full width
- Be a `div` with a class of `cv-layout-body`
- When a body template is specified in content, the template should be rendered inside of `.cv-layout-body`
- When no body template is specified, should default to rendering `ng-content` inside o `.cv-layout-body`

## Sections to be considered

- [Nav](#nav-section)

### Nav section

Thinking for this section is to have a designated place for navigational elements on the application consuming the library.

Since navigation is a very dynamic element in a page, it makes it difficult to make a hard line on where it would be placed. Possibly create `TopNav` and `SideNav` instead of a singular `Nav`... to be considered.

# Suggested process

If you would like to pick up a section to implement, please follow these steps:

> First thing to add... thank you for your effort! It is truly appreciated.

1. Create a new issue `feat(@chill-viking/layout): add {section} section`
2. Assign the new issue to yourself
3. Add a comment on this issue to mention that you have started `{section} section` with a mention of the new issue created
4. Fork repo or create a new branch for the changes
5. Implement change
6. Create a pull request, with mention to this issue and the individual issue
7. Once pull request is merged, mark the section completed in [Primary sections to add](#primary-sections-to-add)

**Comment by peterjokumsen** [comment]
Feel that this objective has been achieved. Will create another issue to focus on navigation elements.
